### ISSUE TITLE
How to obtain Subject Status and Edit Check?

### ISSUE THREAD

**Issue by religou** [mdsol/rwslib#137]
Hi:

The result obtained by StudyVersionRequest is not shown Edit Check Data. How to get Edit Check Data?

And the subject data does not contain Subject Status. How to get it?

Best Regards

**Comment by isparks** [comment]
My memory is getting hazy on this so maybe someone who still works at Medidata could comment but I recall that the ODM Metadata export does not include Edit Check definitions, custom functions or Matrices except the Primary Matrix. I don't think there is a way to get the Edit Checks or Custom Functions. Really, you want to get an Architect Loader Spreadsheet which contains all study object definitions. 

For Subject Status: I advise to check learn.mdsol.com to see if it is documented or to make a comment on it. rwslib provides an interface to what is included in the RWS output by Medidata so if it doesn't contain what you are looking for it would be best to ask Medidata. It might be possible to get Subject Status from the Audit Trail (ODM Adapter) exports but I have not tried it.

**Comment by glow-mdsol** [comment]
Yeah, there was an experiment a very long time ago to work out whether it was possible to have a structure for the edit checks in ODM, but it never went anywhere.  Similarly for Rave specific attributes like CustomFunctions.  The ALS is the best solution here.

I'm not aware of a dataset that includes the status for the subjects but the [StudySubjects](https://learn.medidata.com/en-US/bundle/rave-web-services/page/retrieve_the_list_of_subjects_in_a_study.html) dataset allows you to filter based on the subject status.  It's not wildly efficient, but you could iterate over the available statuses for your study (this is included as the `StudySubjectsRequest` dataset in rwslib.
